### PR TITLE
Spark: Use GCP Dependency + Remove Mode

### PR DESCRIPTION
**Pull request by akhurana001**
# TL;DR

1. Depend on https://github.com/GoogleCloudPlatform/spark-on-k8s-operator multi-version branch, Can't use master since the code currently depends on `PendingSubmissionState` which is only in multi-version CRD.
2. Remove  spark-mode (defaults to cluster in spark-operator) . I believe long-term we potentially want to let platform admins configure this , maybe via spark config.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Depend on https://github.com/GoogleCloudPlatform/spark-on-k8s-operator multi-version branch . Can't use master since the code currently depends on `PendingSubmissionState` which is only in multi-version CRD
Remove mode (defaults to cluster in spark-operator) . I believe long-term we potentially want to let platform admins configure this , maybe via spark config.

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	cloud.google.com/go v0.78.0 // indirect
-	github.com/GoogleCloudPlatform/spark-on-k8s-operator v0.1.3
+	github.com/GoogleCloudPlatform/spark-on-k8s-operator v0.0.0-20200723154620-6f35a1152625
```
Comment (kumare3) [inline go.mod:7]: thats a weird version, i guess we have no choice right now?

**Comment by kumare3** [comment]
This is great thank you, so it seems the upstream just works?

File: go.mod (hunk 1 above)
Comment (akhurana001) [reply go.mod:7]: Yeah, this is the latest sha on https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/tree/multi-version-support .

**Comment by akhurana001** [comment]
@kumare3 @anandswaminathan  this works with multi-version SparkOperator deployed. Also made some minor logging changes. 
Let me know if this looks good ?

**Review by anandswaminathan — DISMISSED** [review]

**Review by kumare3 — APPROVED** [review]
